front/src/components/atoms/ButtonIcon.tsx

```diff
@@ -0,0 +1,19 @@
+import { IconDefinition } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { MouseEventHandler } from "react";
+
+type ButtonIconProps = {
+    onClick: MouseEventHandler<HTMLButtonElement> | (() => void);
```

Comment:
Why do you need two types here? Why not just mouse event handler?

Reply:
I thought that the two were too different, but you are right, casting `() => void` to the event handler works great!